File: packages/flutter/android/src/main/kotlin/dev/mushimushi/flutter/MushiMushiPlugin.kt
package dev.mushimushi.flutter

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/**
 * Flutter plugin scaffolding. The Dart side currently does all the heavy
 * lifting (HTTP, queue, screenshot via RepaintBoundary, shake via
 * sensors_plus). This plugin reserves the `dev.mushimushi.flutter` channel
 * for future native-only capabilities (eg. native screenshot capture, share
 * sheet integration, or native widgets) without breaking the published
 * pub.dev contract.
 */
class MushiMushiPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(binding.binaryMessenger, "dev.mushimushi.flutter")
        channel.setMethodCallHandler(this)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "ping" -> result.success("pong")
            else -> result.notImplemented()
        }
    }
}

File: packages/flutter/ios/Classes/MushiMushiPlugin.swift
import Flutter
import UIKit

/// Flutter plugin scaffolding for iOS. Mirrors the Android plugin: the Dart
/// side does the heavy lifting today, this channel exists for future
/// native-only capabilities (e.g. native screenshot capture, share sheet
/// integration) without breaking the pub.dev contract.
public class MushiMushiPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "dev.mushimushi.flutter",
            binaryMessenger: registrar.messenger())
        let instance = MushiMushiPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "ping":
            result("pong")
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
